src/features/messages/cards/TransactionCard.tsx

```diff
@@ -236,11 +249,43 @@ function CallDataModal({ debugResult }: { debugResult?: MessageDebugResult }) {
           </p>
           <LabelAndCodeBlock label="Recipient contract address:" value={contract} />
           <LabelAndCodeBlock label="Handle function input calldata:" value={handleCalldata} />
+          <button onClick={handleClick}
+            disabled={disabled}
+            className='underline text-blue-400'
+          >
+            {buttonText}
+          </button>
+          {showSpinner && <Spinner classes="mt-4 scale-75 self-center" />}
         </div>
       </Modal>
     </>
   );
 }
+async function simulateCall({contract,handleCalldata,chainId,message}:{contract:string,handleCalldata:string,chainId:ChainId,message:Message}){
+
+  const data={
+        save: true,
+        save_if_fails: true, 
+        simulation_type: 'full',
+        network_id: chainId, 
+        from: '0x0000000000000000000000000000000000000000',//can be any address, doesn't matter
+        to: contract,
+        input:handleCalldata,
+        gas: Number(message.totalGasAmount),
+        gas_price: Number(message.totalPayment)/Number(message.totalGasAmount),
```

Comment:
Use the computeAvgGasPrice util function here. It handles possible rounding issues:
https://github.com/hyperlane-xyz/hyperlane-explorer/blob/main/src/features/messages/cards/GasDetailsCard.tsx#L168